src/sqlfluff/core/dialects/dialect_snowflake.py

```diff
@@ -72,6 +82,9 @@
     ParameterAssignerSegment=KeywordSegment.make(
         "=>", name="parameter_assigner", type="parameter_assigner"
     ),
+    KeywordAssignerSegment=SymbolSegment.make(
+        "=>", name="keyword_assigner", type="keyword_assigner"
+    ),
```

Comment:
I think `ParameterAssignerSegment` and `KeywordAssignerSegment` are actually the same thing and we only need one of them. Previously I've used the term `paramenter` here to avoid confusion with `keyword` in the sql sense rather than `keyword` in the function sense.

Reply:
That being said, I *do* like that you've used `SymbolSegment` rather than `KeywordSegment`! That's definitely an improvement. 👍 

I'd prefer it if we just had one of these two, my vote would be for `ParameterAssignerSegment`, but I do see the logic for `KeywordAssignerSegment`, and it has been somewhat demonstrated by you creating a new segment rather than using the old one. I'm happy for you to decide which we keep, I just think there should be only one!